bokeh/charts/_builder.py

```diff
@@ -258,64 +221,4 @@ def set_and_get(self, prefix, val, content):
         """
         self._set_and_get(self.data, prefix, self.attr, val, content)
 
-    @property
-    def palette(self):
```

Comment:
I was planning on adding a cycle_palette property to the Builder hasprops base (that can just be set by users) and have cycle_colors use this by default everywhere. Does that sound ok?

I like this better because it leaves more control to builder internals in case there is more than one set of groups. 
 


> On Feb 5, 2015, at 03:16, fpliger <notifications@github.com> wrote:
> 
> In bokeh/charts/_builder.py:
> 
> > @@ -258,64 +221,4 @@ def set_and_get(self, prefix, val, content):
> >          """
> >          self._set_and_get(self.data, prefix, self.attr, val, content)
> >  
> > -    @property
> > -    def palette(self):
> Ah, palette property was definitely something Ieft for later in the refactoring that I forgot (grrr, sorry). The idea was to remove _set_colors in favor of palette.
> 
> Now, if I got it write, you've added cycle_colors which is used on all builders and palette is not used anyway. As palette is set on the Builder.__init_ what about:
> 
> changing _palette to palette
> change self._palette = palette or DEFAULT_PALETTE to self.palette = palette or cycle_colors(...)
> make all custom builders use self.palette instead of cycle_colors directly
> ?
> 
> —
> Reply to this email directly or view it on GitHub.
>